### DISCUSSION TITLE
Splitting connected ROMs (S3&K, from Sega Mega Drive/Genesis Classics)

### DISCUSSION THREAD

**Discussion by TheNametag** [farmerbb/RED-Project#76] [General]
Is there a known way to separate the Sonic 3 and Sonic & Knuckles individual ROMs from the combined ROM found in the Sega Mega Drive & Genesis Classics?  For a few reasons.  First, to have the original versions available, and second in order to then combine Sonic & Knuckles with the other games to get Knuckles in Sonic 2, and Blue Sphere.

**Comment by farmerbb** [comment]
The `Sonic_Knuckles_wSonic3.bin` file is just the Sonic & Knuckles ROM with the Sonic 3 ROM appended to the end of it.  On Linux you can split the ROMs by running these commands:

```
split -b 2M Sonic_Knuckles_wSonic3.bin
mv xaa "Sonic & Knuckles.bin"
mv xab "Sonic the Hedgehog 3.bin"
```

On Windows you can use 7-Zip to split the file, just browse to the `Sonic_Knuckles_wSonic3.bin` file inside 7-Zip, right click -> Split file... -> enter "2M" in the "Split to volumes, bytes:" field.  The split file ending in 001 is the Sonic & Knuckles ROM, and the one ending in 002 is the Sonic 3 ROM.

**Comment by farmerbb** [comment]
Also, while you'll be able to recombine Sonic & Knuckles with a Sonic 1 ROM to get Blue Sphere, sadly you won't be able to recombine it with a Sonic 2 ROM to get Knuckles in Sonic 2 as that game requires additional ROM data present on the Sonic & Knuckles cartridge but not present in the ROM found in Sega Genesis Classics on Steam.

**Reply by kjetil-f** [reply]
When I combine the Sonic & Knuckles (with 3 removed) with Sonic 1 from Steam, I only get the No Way screen.

**Reply by RealRelativeEase** [reply]
I might be too late by now, but just in case: You have to press all three face buttons at once on the No Way screen.

**Comment by TheNametag** [comment]
Thanks for the info.  I was able to get the ROMs split, and get Blue Sphere.  Shame Knuckles in Sonic 2 isn't obtainable that way.

**Comment by PW5190** [comment]
The standalone Sonic 3 ROM when separated from Sonic & Knuckles is actually the Japanese version, just like with the uncompressed ROM for Phantasy Star IV.

**Reply by farmerbb** [reply]
For Sonic 3 the only difference between regional variants is the region code in the header. There are no other changes in the ROM

**Comment by SlashEraser** [comment]
[Sonic & Knuckles Lock-On.zip](https://github.com/user-attachments/files/30713032/Sonic.Knuckles.Lock-On.zip)

Drag the S3&K ROM onto s3k_separator.exe, and you will get the Sonic 3 and Sonic & Knuckles ROMs. Rename the extension to .bin.
Then, use Sonic & Knuckles Lock-On ROM Creator V1 or V2 to combine Sonic & Knuckles with Sonic and Sonic 2 to get Blue Sphere and Knuckles in Sonic 2.


<img width="475" height="508" alt="bs" src="https://github.com/user-attachments/assets/557d3cff-dbfc-492f-b1f3-3071ddd24f6b" />

<img width="475" height="508" alt="ks2" src="https://github.com/user-attachments/assets/cae7388f-91d4-4e92-b39a-16b510a764e0" />
